denylist/src/denylist.rs

```diff
@@ -110,8 +104,42 @@ impl DenyList {
     }
 }
 
+/// deconstruct bytes into the filter component parts
+pub fn filter_from_bin(bin: &Vec<u8>) -> Result<Xor32> {
+    if bin.is_empty() {
+        tracing::debug!("empty filter bin");
+        return Err(Error::InvalidBinary("empty filter bin".to_string()));
+    }
+    let mut buf: &[u8] = bin;
+    let _version = buf.get_u8();
+    let signature_len = buf.get_u16_le() as usize;
+    let signature = buf.copy_to_bytes(signature_len).to_vec();
+    let pubkey = PublicKey::from_str(PUB_KEY_B58).expect("failed to decode pub key");
```

Comment:
Find all your uses of expect and get rid of them

Reply:
what so bad with expect ? I see it in use across all projects in this workspace